extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessor.java

```diff
@@ -103,13 +107,17 @@ public WindowOperatorQueryFrameProcessor(
     this.frameWriterFactory = frameWriterFactory;
     this.operatorFactoryList = operatorFactoryList;
     this.jsonMapper = jsonMapper;
-    this.frameReader = frameReader;
     this.query = query;
     this.frameRowsAndCols = new ArrayList<>();
     this.resultRowAndCols = new ArrayList<>();
     this.objectsOfASingleRac = new ArrayList<>();
     this.maxRowsMaterialized = maxRowsMaterializedInWindow;
     this.partitionColumnNames = partitionColumnNames;
+
+    this.frameReader = frameReader;
+    for (int i = 0; i < frameReader.signature().size(); i++) {
+      typeStrategies.add(frameReader.signature().getColumnType(i).get().getNullableStrategy());
```

Comment:
nit: can check if the column type is known. Though it likely is getting verified upstream.

Reply:
I haven't stumbled upon a case where the column type wasn't present, yet. I thought I'll add an explicit check if we stumble upon such a case in this layer in the future.